### PR TITLE
Add IncomingMessageProperties injection support post

### PR DESCRIPTION
**Pull request by g7ed6e**

### CODE REVIEW

File: _posts/2024-05-06-message-property-injection.markdown [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+# Layout
+layout: post
+title:  "Add IncomingMessageProperty injection support anf Kafka transport enhancements in CoreWCF"
+date:   2024-05-06 13:00:00 +0200
+categories: release
+# Author
+author: Guillaume Delahaye (https://github.com/g7ed6e)
+---
+### Introduction
+The CoreWCF 1.6.0 release introduced a new feature that allows to inject incoming message properties as operation contract parameter.
```
Comment (mconnew) [inline _posts/2024-05-06-message-property-injection.markdown]:
```suggestion
The CoreWCF 1.6.0 release introduced a new feature that enables injecting incoming message properties as operation contract parameters.
```

File: _posts/2024-05-06-message-property-injection.markdown [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+{
+    ServiceModel.Channels.KafkaMessageProperty outgoingProperty = new();
+    outgoingProperty.Headers.Add(new ServiceModel.Channels.KafkaMessageHeader("header1", Encoding.UTF8.GetBytes("header1Value")));
+    outgoingProperty.PartitionKey = Encoding.UTF8.GetBytes("key");
+    System.ServiceModel.OperationContext.Current.OutgoingMessageProperties[ServiceModel.Channels.KafkaMessageProperty.Name] =
+        outgoingProperty;
+    channel.DoSomething();
+}
+```
+Service side the implementer can get these values back by pulling the `KafkaLessageProperty`
```
Comment (mconnew) [inline _posts/2024-05-06-message-property-injection.markdown]:
```suggestion
Service side the implementer can get these values back by injecting the `KafkaMessageProperty`
```

File: _posts/2024-05-06-message-property-injection.markdown [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+public class HelloWorldService : IHelloService
+{
+    public string SayHello([Injected(PropertyName = RemoteEndpointMessageProperty.Name)] RemoteEndpointMessageProperty remoteEndpointMessageProperty)
+    {
+        return $"Hello from {remoteEndpointMessageProperty.Address}:{remoteEndpointMessageProperty.Port}";
+    }
+}
+```
+### BasicHttpBinding usage
+A service exposing a `BasicHttpBinding` can pull the `HttpRequestMessageProperty`.
```
Comment (mconnew) [inline _posts/2024-05-06-message-property-injection.markdown]:
```suggestion
A service exposing a `BasicHttpBinding` can inject the `HttpRequestMessageProperty`.
```

File: _posts/2024-05-06-message-property-injection.markdown [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public sealed class InjectedAttribute : Attribute
+    {
+        public string PropertyName { get; set; }
+    }
+}
+```
+### NetTcpBinding usage
+
+A service exposing a `NetTcpBinding` can pull the `RemoteEndpointMessageProperty`.
```
Comment (mconnew) [inline _posts/2024-05-06-message-property-injection.markdown]:
```suggestion
A service exposing a `NetTcpBinding` can inject the `RemoteEndpointMessageProperty`.
```

File: _posts/2024-05-06-message-property-injection.markdown [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+date:   2024-05-06 13:00:00 +0200
+categories: release
+# Author
+author: Guillaume Delahaye (https://github.com/g7ed6e)
+---
+### Introduction
+The CoreWCF 1.6.0 release introduced a new feature that allows to inject incoming message properties as operation contract parameter.
+
+### Implementation
+This feature is implemented in the `OperationContractParameterGenerator` which already support injecting service registered in the DI container or `HttpContext` into the operation contract parameters.
```
Comment (mconnew) [inline _posts/2024-05-06-message-property-injection.markdown]:
```suggestion
This feature is implemented in the `OperationContractParameterGenerator` which already support injecting services registered in the DI container or `HttpContext` into the operation contract parameters.
```

File: _posts/2024-05-06-message-property-injection.markdown [hunk 6]
```diff
@@ -0,0 +1,124 @@
+---
+# Layout
+layout: post
+title:  "Add IncomingMessageProperty injection support anf Kafka transport enhancements in CoreWCF"
+date:   2024-05-06 13:00:00 +0200
```
Comment (mconnew) [inline _posts/2024-05-06-message-property-injection.markdown]: Update the date to something more current.

Comment (g7ed6e) [reply _posts/2024-05-06-message-property-injection.markdown]: Updated!

**Comment by mconnew** [comment]
Closing as replaced by new PR which modifies the blog post date.

**Pull request closed without merging**